bundle/src/main/java/com/adobe/acs/commons/rewriter/impl/ContentVariableTransformer.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2021 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.rewriter.impl;
+
+import com.adobe.acs.commons.ccvar.PropertyAggregatorService;
+import com.adobe.acs.commons.ccvar.PropertyConfigService;
+import com.adobe.acs.commons.ccvar.util.ContentVariableReplacementUtil;
+import com.adobe.acs.commons.rewriter.ContentHandlerBasedTransformer;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.rewriter.ProcessingComponentConfiguration;
+import org.apache.sling.rewriter.ProcessingContext;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.AttributesImpl;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * {@link org.apache.sling.rewriter.Transformer} used to process HTML requests and replace content tokens found in the
+ * rendered HTML.
+ */
+public class ContentVariableTransformer extends ContentHandlerBasedTransformer {
+
+    private Map<String, Object> contentVariableReplacements;
+    private PropertyAggregatorService aggregatorService;
+    private PropertyConfigService propertyConfigService;
+
+    public ContentVariableTransformer() {
+    }
+
+    public ContentVariableTransformer(PropertyAggregatorService propertyAggregatorService,
+                                      PropertyConfigService propertyConfigService) {
+        this.aggregatorService = propertyAggregatorService;
+        this.propertyConfigService = propertyConfigService;
+    }
+
+    @Override
+    public void init(ProcessingContext processingContext, ProcessingComponentConfiguration processingComponentConfiguration) throws IOException {
+        SlingHttpServletRequest request = processingContext.getRequest();
+
+        contentVariableReplacements = aggregatorService.getProperties(request);
+    }
+
+    public void startElement(String uri, String localName, String quaName, Attributes atts) throws SAXException {
+        if (shouldRun()) {
+            AttributesImpl newAttrs = new AttributesImpl(atts);
+            for (int i = 0; i < newAttrs.getLength(); i++) {
+                String currentAttribute = newAttrs.getValue(i);
+                final List<String> keys = ContentVariableReplacementUtil.getKeys(currentAttribute);
```

Comment:
Need to check if `currentAttribute` is null here - there are times where it can be null and that triggers a NPE that breaks the page rendering.

Reply:
added a check on the attribute!